fix: vue template syntax bug #503

nova1751 opened the pull request:
- [x] Format all commit messages with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] This PR fixes #503

This bug is caused by fact that the `vue.tmLanguage.json` in [`Volar` repo](https://github.com/vuejs/language-tools/tree/master/extensions/vscode/syntaxes) doesn't support Vue directives and interpolations,the Volar uses `Injections` instead.The PR use `vue-html` syntax to render the code in tag `template` to fix the bug.

- the original：
  ![Snipaste_2023-11-28_20-27-18](https://github.com/shikijs/shiki/assets/97817985/54c11552-eb0d-425a-bdd8-3a5be82118ad)
- use `vue-html`： 
  ![Snipaste_2023-11-28_20-29-05](https://github.com/shikijs/shiki/assets/97817985/d121ad34-4a72-4069-a746-86710f9f8832)

antfu commented:
The grammar is automatically fetched from upstream, so changing them will not fix it. Instead it's that Shiki is currently lacking for the injection support, which I fixed implemented it in Shikiji: https://github.com/antfu/shikiji/pull/48

nova1751 commented:
> The grammar is automatically fetched from upstream, so changing them will not fix it. Instead it's that Shiki is currently lacking for the injection support, which I fixed implemented it in Shikiji: [antfu/shikiji#48](https://github.com/antfu/shikiji/pull/48)

Well,I get it.I fixed this thing by replacing the normal `html`'s rendering logic with the `vue-html`'s which could make elements render correctly in tag `template`.It works without injections actually,though it might be a bit weird to change the files fetched automatically.
